src/tests/strategies/polygon/balancer/testBalancerBase.js

```diff
@@ -0,0 +1,151 @@
+
+      expect(_after).to.be.gt(_before, "picklejar withdrawAll failed");
+      expect(_after).to.be.gt(_want, "no interest earned");
+    });
+
+    it("Should harvest correctly", async () => {
+      const _want = await want.balanceOf(alice.address);
+      await want.approve(pickleJar.address, _want);
+      await pickleJar.deposit(_want);
+      console.log("Alice pTokenBalance after deposit: %s\n", (await pickleJar.balanceOf(alice.address)).toString());
+      await pickleJar.earn();
+
+      const _before = await pickleJar.balance();
+      let _treasuryBefore = await want.balanceOf(treasury.address);
+
+      console.log("Picklejar balance before harvest: ", _before.toString());
+      console.log("💸 Treasury balance before harvest: ", _treasuryBefore.toString());
+      console.log("\nRatio before harvest: ", (await pickleJar.getRatio()).toString());
+
+      await rewardToken.transfer(strategy.address, toWei(1));
+
+      await strategy.harvest();
+
+      const _after = await pickleJar.balance();
+      let _treasuryAfter = await want.balanceOf(treasury.address);
+      console.log("Ratio after harvest: ", (await pickleJar.getRatio()).toString());
+      console.log("\nPicklejar balance after harvest: ", _after.toString());
+      console.log("💸 Treasury balance after harvest: ", _treasuryAfter.toString());
+
+      //20% performance fee is given
+      const earned = _after.sub(_before).mul(1000).div(800);
+      const earnedRewards = earned.mul(200).div(1000);
+      const actualRewardsEarned = _treasuryAfter.sub(_treasuryBefore);
+      console.log("\nActual reward earned by treasury: ", actualRewardsEarned.toString());
+
+      expect(earnedRewards).to.be.eqApprox(actualRewardsEarned, "20% performance fee is not given");
+
+      //withdraw
+      const _devBefore = await want.balanceOf(devfund.address);
+      _treasuryBefore = await want.balanceOf(treasury.address);
+      console.log("\n👨‍🌾 Dev balance before picklejar withdrawal: ", _devBefore.toString());
+      console.log("💸 Treasury balance before picklejar withdrawal: ", _treasuryBefore.toString());
+
+      const _cooldown = await pickleJar.balanceOf(alice.address);
+
+      await pickleJar.cooldown(_cooldown);
+
+      await increaseTime(60 * 60 * 24 * 3); //travel 3 days
+
+      await pickleJar.withdrawAll();
+
+      console.log("_after: ", _after.toString());
+
+      const expectedDevFund = _after.mul(50).mul(4).div(1000).div(7);
+
+      const _devAfter = await want.balanceOf(devfund.address);
+      _treasuryAfter = await want.balanceOf(treasury.address);
+      console.log("\n👨‍🌾 Dev balance after picklejar withdrawal: ", _devAfter.toString());
+      console.log("💸 Treasury balance after picklejar withdrawal: ", _treasuryAfter.toString());
+
+      //0% goes to treasury
+      const _treasuryFund = _treasuryAfter.sub(_treasuryBefore);
+      expect(_treasuryFund).to.be.eq(0, "treasury've stolen money!!!!");
```

Comment:
You have a `_devBefore` and a `_devAfter`. I assume you meant to include an assertion that the difference is equal to `expectedDevFund`, correct?

Reply:
correct, updated it